core/src/test/java/org/apache/accumulo/core/client/admin/NewTableConfigurationTest.java

```diff
@@ -93,4 +105,57 @@ public void testCreateOffline() {
     NewTableConfiguration ntcOnline = new NewTableConfiguration();
     assertTrue(ntcOnline.getInitialTableState() == InitialTableState.ONLINE);
   }
+
+  public void populateOptions() {
+    options = new HashMap<>();
+    options.put("hasher", "murmur3_32");
+    options.put("modulus", "5");
+  }
+
+  /**
+   * Verify enableSampling returns
+   */
+  @Test
+  public void testEnableSampling() {
+    SamplerConfiguration sha1SamplerConfig = new SamplerConfiguration("com.mysampler");
+    sha1SamplerConfig.setOptions(options);
+    NewTableConfiguration ntcSample2 =
+        new NewTableConfiguration().enableSampling(sha1SamplerConfig);
+    assertEquals("com.mysampler", ntcSample2.getProperties().get("table.sampler"));
+    assertEquals("5", ntcSample2.getProperties().get("table.sampler.opt.modulus"));
+    assertEquals("murmur3_32", ntcSample2.getProperties().get("table.sampler.opt.hasher"));
+  }
+
+  /**
+   * Verify enableSummarization returns SummarizerConfiguration with the expected class name(s).
+   */
+  @Test
+  public void testEnableSummarization() {
+    SummarizerConfiguration summarizerConfig1 = SummarizerConfiguration
+        .builder("com.test.summarizer").setPropertyId("s1").addOption("opt1", "v1").build();
+    NewTableConfiguration ntcSummarization1 =
+        new NewTableConfiguration().enableSummarization(summarizerConfig1);
+
+    Map<String,String> props = ntcSummarization1.getProperties();
+    assertEquals("v1", props.get("table.summarizer.s1.opt.opt1"));
+    assertEquals("com.test.summarizer", props.get("table.summarizer.s1"));
+
+    Class<? extends Summarizer> builderClass = FamilySummarizer.class;
+    assertTrue(Summarizer.class.isAssignableFrom(builderClass));
+
+    SummarizerConfiguration summarizerConfig2 = SummarizerConfiguration.builder(builderClass)
+        .setPropertyId("s2").addOption("opt2", "v2").build();
+    NewTableConfiguration ntcSummarization2 =
+        new NewTableConfiguration().enableSummarization(summarizerConfig2);
+    assertEquals(builderClass.getName(),
+        ntcSummarization2.getProperties().get("table.summarizer.s2"));
+
+    NewTableConfiguration ntcSummarization3 =
+        new NewTableConfiguration().enableSummarization(summarizerConfig1, summarizerConfig2);
+    assertEquals("com.test.summarizer",
+        ntcSummarization3.getProperties().get("table.summarizer.s1"));
+    assertEquals(builderClass.getName(),
+        ntcSummarization3.getProperties().get("table.summarizer.s2"));
+
```

Comment:
Could add checks to ensure the options `table.summarizer.s1.opt.opt1=v1` and `table.summarizer.s2.opt.opt2=v2` are correct.